### PR TITLE
Fix build breaks with gcc 11.3.1

### PR DESCRIPTION
**Pull request by dsouzai**
* Fix documentation to ensure `OMR_GC_MODRON_SCAVENGER` is enabled during the build
* Add `-O3` to `omrtrace` component to prevent warning about not compiling with `-O` when `_FORTIFY_SOURCE` is specified.

### CODE REVIEW

**Comment by dsouzai** [comment]
@keithc-ca @babsingh could you please review?

**Comment by babsingh** [comment]
We should be able to build OMR without `OMR_GC_MODRON_SCAVENGER`. It should not be a mandatory requirement. So, the below error should be fixed by either consistently using the same ifdef macros or enabling dependent ifdef macros correctly. `MarkingScheme.hpp:292` uses `OMR_GC_CONCURRENT_SCAVENGER` whereas `GCExtensionsBase.hpp` uses `OMR_GC_MODRON_SCAVENGER` for `isScavengerBackOutFlagRaised`. fyi @dmitripivkine @amicic @LinHu2016 

```
/home/harpreet/omr/gc/base/MarkingScheme.hpp:292:67: error: 'class MM_GCExtensionsBase' has no member named 'isScavengerBackOutFlagRaised'; did you mean 'isScavengerEnabled'?
   if (_extensions->isConcurrentScavengerEnabled() && _extensions->isScavengerBackOutFlagRaised()) {
                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                                   isScavengerEnabled
```

**Comment by dmitripivkine** [comment]
> We should be able to build OMR without `OMR_GC_MODRON_SCAVENGER`. It should not be a mandatory requirement. So, the below error should be fixed by either consistently using the same ifdef macros or enabling dependent ifdef macros correctly. `MarkingScheme.hpp:292` uses `OMR_GC_CONCURRENT_SCAVENGER` whereas `GCExtensionsBase.hpp` uses `OMR_GC_MODRON_SCAVENGER` for `isScavengerBackOutFlagRaised`. fyi @dmitripivkine @amicic @LinHu2016
> 
> ```
> /home/harpreet/omr/gc/base/MarkingScheme.hpp:292:67: error: 'class MM_GCExtensionsBase' has no member named 'isScavengerBackOutFlagRaised'; did you mean 'isScavengerEnabled'?
>    if (_extensions->isConcurrentScavengerEnabled() && _extensions->isScavengerBackOutFlagRaised()) {
>                                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
>                                                                    isScavengerEnabled
> ```

Yes, if should be fixed eventually. However I have suspicion the condition check was missed at time of split code for OMR many years ago. I think there are more missed checks like this in the code (not only Scavenger but another GC features related). So OMR code requires inspection/cleaning project. Also when it completes it required testing support - compiling code with disabled features on regular basis. Otherwise new errors are going to be introduced unavoidably. Currently we have no resources to address this, so, this is another issue for backlog

**Comment by babsingh** [comment]
LGTM. GC fix is good temporary workaround. `-O3` is only set when `_FORTIFY_SOURCE` is enabled on `Linux`. @keithc-ca Do these change look good to you?

File: README.md [hunk 1]
```diff
@@ -145,7 +145,7 @@ source tree. For more detailed instructions please read [BuildingWithCMake.md](d
     cd build
 
     # Generate the build system using cmake
-    cmake ..
+    cmake -Wdev -C../cmake/caches/Travis.cmake  ..
```
Comment (keithc-ca) [inline README.md:148]: nit: The [documentation for cmake](https://cmake.org/cmake/help/latest/manual/cmake.1.html) suggests `-C` and the path are expected to be separate command-line arguments.

Comment (dsouzai) [reply README.md]: Oh, I guess I just didn't notice that.

Comment (dsouzai) [reply README.md]: Fixed.

**Review by babsingh — APPROVED** [review]

**Comment by babsingh** [comment]
jenkins build all

To make sure, the `-O3` change works fine on all Linux platforms.

**Review by keithc-ca — APPROVED** [review]

**Comment by babsingh** [comment]
> https://ci.eclipse.org/omr/job/PullRequest-win_x86-64/3469/console

@AdamBrousseau Noticed a new infra issue on Windows, which is unrelated to this PR. Is this a known infra issue?

**Comment by AdamBrousseau** [comment]
jenkins build win

**Comment by AdamBrousseau** [comment]
We were testing a new machine. Back to normal now.

**Comment by dsouzai** [comment]
riscv build failed because of:
```
11:23:11  20: [----------] 18 tests from RWMutex
13:05:04  Cancelling nested steps due to timeout
13:05:04  Sending interrupt signal to process
13:05:24  After 20s process did not stop
```
It's not the first time there's been a hang in RWMutex (see https://github.com/eclipse/omr/pull/277#issuecomment-250488086) though that was all the way in 2016 on OSX, so kind of a stretch to link these two.

Will run the test again to see if it's reproducable.

**Comment by dsouzai** [comment]
jenkins build riscv64

**Comment by dsouzai** [comment]
Don't see the RWMutex hang this time around; riscv64 failure now because of https://github.com/eclipse/omr/issues/6704

**Comment by babsingh** [comment]
The failures seen in the PR builds are unrelated to the changes in this PR: https://github.com/eclipse/omr/issues/6704 and https://github.com/eclipse/omr/issues/6706. Thus, merging.

**Pull request merged**